### PR TITLE
Use processor instances instead classes

### PR DESCRIPTION
**Pull request by Exelord**

### CODE REVIEW

File: src/processors/operation-processor.ts [hunk 1]
```diff
@@ -3,13 +3,14 @@ import Resource from "../resource";
 import { Operation, ResourceConstructor } from "../types";
 
 export default class OperationProcessor<ResourceT = Resource> {
-  public app: Application;
```
Comment (rtablada) [inline src/processors/operation-processor.ts]: Why did you remove Application?

File: src/processors/operation-processor.ts [hunk 2]
```diff
@@ -3,13 +3,14 @@ import Resource from "../resource";
...
-  public resourceClass?: ResourceConstructor;
+  static resourceClass?: ResourceConstructor;
 
-  async shouldHandle(op: Operation): Promise<boolean> {
+  static async shouldHandle(op: Operation): Promise<boolean> {
     return this.resourceClass && op.ref.type === this.resourceClass.type;
   }
 
+  constructor(protected app: Application) {}
```
Comment (rtablada) [inline src/processors/operation-processor.ts]: Should this constructor set this.app?

File: src/processors/operation-processor.ts (hunk 1 above)
Comment (Exelord) [reply src/processors/operation-processor.ts]: Move to the constructor. Previously we iterating through all processors instances and injecting the app directly on the object like` processor.app = this` in application constructor. Right now we dont have to do it as we have an access to the processor initializer

File: src/processors/operation-processor.ts (hunk 2 above)
Comment (Exelord) [reply src/processors/operation-processor.ts]: >Moved to the constructor. Previously we iterating through all processors instances and injecting the app directly on the object like processor.app = this in application constructor. Right now we dont have to do it as we have an access to the processor initializer

File: src/processors/operation-processor.ts (hunk 1 above)
Comment (rtablada) [reply src/processors/operation-processor.ts]: Don't declare properties in the constructor, don't get clever. Use `protected` in the class definition and then set the value in the constructor.

Comment (Exelord) [reply src/processors/operation-processor.ts]: Ok, Im new to typescript and thought that this is the way now as I saw that in TS doc
